File: Daily/PY/Leetcode989-数组形式的加法.py
from typing import List
class Solution:
    def addToArrayForm(self, A: List[int], K: int) -> List[int]:
        res = list()
        n = len(A)
        for i in range(n-1,-1,-1):
            sum = A[i] + K % 10
            #地板除
            K = K // 10
            if sum >= 10:
                sum = sum - 10
                #把carry位放到所有数K里
                K = K + 1
            res.append(sum)
        while K > 0:
            res.append(K % 10)
            K = K // 10
        res.reverse()
        return res

File: Daily/sbgo/Leetcode989-数组形式的加法.go
package sbgo

func addToArrayForm(A []int, K int) []int {
	n := len(A)
	res := []int{}
	for i:=n-1;i >= 0;i--{
		sum := A[i] + K %10
		K = K / 10
		if sum >= 10{
			sum -= 10
			K++
		}
		res = append(res,sum)
	}
	for K > 0{
		res = append(res,K % 10)
		K = K / 10
	}
	reverse(res)
	return res
}

func reverse(A []int) {
    for i, n := 0, (len(A)-1); i <= n; {
        A[i], A[n] = A[n], A[i]
        i++
        n--
    }
}
